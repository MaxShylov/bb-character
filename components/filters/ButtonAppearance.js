import { useState, useCallback } from 'react';
import { Button, StyleSheet, View } from 'react-native';

export const ButtonAppearance = ({ title = '', onPress }) => {
  const [active, setActive] = useState();

  const handlePressButton = useCallback(() => {
    const nextState = !active
    setActive(nextState)
    onPress && onPress(nextState)
  }, [active, onPress])

  return (
    <View style={[styles.buttonWrapper, active && styles.active]}>
      <Button style={styles.button} title={title} onPress={handlePressButton} />
    </View>
  )
}

const styles = StyleSheet.create({
  buttonWrapper: {
    width: 40,
    height: 40,
    borderRadius: 7,
    marginLeft: 10,
    alignItems: 'center',
    justifyContent: 'center',
  },
  active: {
    backgroundColor: '#a3c8fd',
  },
})
