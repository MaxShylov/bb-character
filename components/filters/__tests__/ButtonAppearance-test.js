import renderer from 'react-test-renderer';
import { ButtonAppearance } from '../ButtonAppearance';

test('renders correctly', () => {
  const tree = renderer.create(<ButtonAppearance />).toJSON();

  expect(tree).toMatchSnapshot();
});
