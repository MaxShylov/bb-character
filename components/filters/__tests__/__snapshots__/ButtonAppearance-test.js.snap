// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
        "borderRadius": 7,
        "height": 40,
        "justifyContent": "center",
        "marginLeft": 10,
        "width": 40,
      },
      undefined,
    ]
  }
>
  <View
    accessibilityRole="button"
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "margin": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;
